packages/parameters/tests/e2e/secretsProvider.class.test.functionCode.ts

```diff
@@ -0,0 +1,57 @@
+import { Context } from 'aws-lambda';
+import { SecretsProvider } from '../../lib/secrets';
+import { TinyLogger } from '../helpers/tinyLogger';
+import { transform } from 'esbuild';
+import { SecretsGetOptionsInterface } from '../../lib/types';
+
+const logger = new TinyLogger();
+const provider = new SecretsProvider();
+
+const secretNamePlain = process.env.SECRET_NAME_PLAIN || '';
+const secretNameObject = process.env.SECRET_NAME_OBJECT || '';
+const secretNameBinary = process.env.SECRET_NAME_BINARY || '';
+const secretNameObjectWithSuffix = process.env.SECRET_NAME_OBJECT_WITH_SUFFIX || '';
+const secretNameBinaryWithSuffix = process.env.SECRET_NAME_BINARY_WITH_SUFFIX || '';
+
+const _call_get = async (paramName: string, testName: string, options?: SecretsGetOptionsInterface) : Promise<void> => {
```

Comment:
How will this work once we introduce multiple providers (i.e. once #1222 is merged and we need multiple SecretsProviders)? Should we add a new parameter?

Reply:
Right, lets pass the provider in the signature. Should we wait until #1222 is merged, or can I change it now?

Reply:
Let's do it now, so later we can just add a new block at the end of the main handler

Reply:
Hey guys! #1222 is ready for review.